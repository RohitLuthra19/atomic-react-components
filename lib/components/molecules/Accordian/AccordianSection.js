import { PureComponent } from 'react';
import type { Props, State } from './types';

class AccordionSection extends PureComponent<Props, State> {
  onClick = () => {
    this.props.onClick(this.props.label);
  };

  render() {
    const {
      onClick,
      props: { isOpen, label, className },
    } = this;

    return (
      <div>
        <summary onClick={onClick} className={className}>
          {label}
        </summary>
        {isOpen ? <div className="accordianBody">{this.props.children}</div> : null}
      </div>
    );
  }
}

export default AccordionSection;
