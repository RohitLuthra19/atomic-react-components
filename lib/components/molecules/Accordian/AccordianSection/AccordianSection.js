// @flow
/**
 *
 * AccordianSection
 *
 */
import { PureComponent } from 'react';
import styled from 'styled-components';
import styles from './AccordianSection.style';
import type { Props } from './types';

class AccordianSection extends PureComponent<Props> {
  onClick = () => {
    this.props.onClick(this.props.label);
  };

  render() {
    const {
      onClick,
      props: { isOpen, header, children },
    } = this;

    return (
      <div className="accordian">
        <summary
          onClick={onClick}
          aria-expanded={isOpen}
          className="accordian-item"
          ref={this.accordianRef}
        >
          {header}
        </summary>
        {isOpen ? <div className="accordian-panel">{children}</div> : null}
      </div>
    );
  }
}

export default styled(AccordianSection)`
  ${styles};
`;

export { AccordianSection as AccordianSectionVanilla };
