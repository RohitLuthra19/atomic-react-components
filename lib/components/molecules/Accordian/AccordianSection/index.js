// @flow
export { default } from './AccordianSection';
export { AccordianSectionVanilla } from './AccordianSection';
