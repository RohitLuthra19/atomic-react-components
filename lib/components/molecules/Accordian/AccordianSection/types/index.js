// @flow

export type Props = {
  children: Array<any>,
  isOpen: boolean,
  label: string,
  header: object,
};
