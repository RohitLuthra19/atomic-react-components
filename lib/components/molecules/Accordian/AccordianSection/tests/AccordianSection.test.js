import { mount } from 'enzyme';
// import renderer from 'react-test-renderer';

import AccordianSection from '../index';

describe('<AccordianSection />', () => {
  let AccordianComponent;

  beforeEach(() => {
    AccordianComponent = mount(<AccordianSection />);
  });

  test('Accordian should collapse correctly', () => {
    expect(AccordianComponent.find('.accordian-panel').length).toEqual(0);
  });
});
