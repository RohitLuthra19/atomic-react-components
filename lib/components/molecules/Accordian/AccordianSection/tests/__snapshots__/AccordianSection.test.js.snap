// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordianSection /> should render correctly 1`] = `
<AccordianSection
  className="AccordianSection-sc-49ro5k-0 fjTVuV"
>
  Test
</AccordianSection>
`;
